core/src/main/java/org/apache/calcite/sql/fun/SqlItemOperator.java

```diff
@@ -89,6 +88,9 @@
     }
     final RelDataType operandType = callBinding.getOperandType(0);
     final SqlSingleOperandTypeChecker checker = getChecker(operandType);
+    if (checker == null) {
+      throw callBinding.newValidationSignatureError();
```

Comment:
Thank you very much for the change, @pengzhiwei2018!

But why using a null check rather than changing the default case in switch block? Seems they are effectively equivalent. Sorry to be a little picky ;)

Reply:
Thanks very much for you suggestions @zhztheplayer ! I have updated the code.